lib/cucumber/core/report/summary.rb

```diff
@@ -4,20 +4,22 @@ module Report
       class Summary
         attr_reader :test_cases, :test_steps
 
-        def initialize
+        def initialize(event_bus)
           @test_cases = Test::Result::Summary.new
           @test_steps = Test::Result::Summary.new
+          subscribe_to(event_bus)
```

Comment:
If this class is connected directly to the `event_bus` then it need to filter out the notification from the executed `hooks`, since they are not included in the step counts.